src/app/core/cart/cart.service.ts

```diff
@@ -61,12 +66,16 @@ export class CartService {
 
     private evaluateCartWeight(items: any): number {
         return items
-            .filter((item: any) => item?.dimensions?.weight != null && item?.quantity != null)
-            .reduce((sum: number, item: any) => sum + (item.dimensions.weight * item.quantity), 0);
+            .map((item: any) => item.quantity * this.getSnipcartItemWeight(item))
+            .reduce((sum: number, weight: number) => sum + weight, 0);
+    }
+
+    private getSnipcartItemWeight(item: any): number {
+        let weightField = item.customFields.find((field: any) => field.name == "Weight");
+        return weightField?.value != null ? this.weightPipe.transform(weightField.value) : 0;
```

Comment:
are there instances when the value will be null when weightField is not null?

I am not sure, but I think just doing weightField ? this.weightPipe... : 0 is sufficient. I am not 100%, there may be some edge case that I am not thinking of